engine/common/synchronization/request_handler.go

```diff
@@ -266,7 +272,10 @@ func (r *RequestHandlerEngine) onBatchRequest(originID flow.Identifier, req *mes
 
 	// deduplicate the block IDs in the batch request
 	blockIDs := make(map[flow.Identifier]struct{})
-	for _, blockID := range req.BlockIDs {
+	for i, blockID := range req.BlockIDs {
+		if i == int(r.core.(*synchronization.Core).Config.MaxSize) {
+			break
+		}
```

Comment:
Is a client susceptible to sending a request for a small amount of BlockIDs that is so full of duplicates that it appears to be `> r.core.(*synchronization.Core).Config.MaxSize`?

Could we instead check `len(blockIDs)` as an additional condition to entering the loop?